### PR TITLE
Managed swiftlint integration

### PR DESCRIPTION
**Pull request by thiagofelix**
PR inspired by issues #23 and #25

As of now `danger-swiftlint` doesn't know which version of `swiftlint` is installed on the host computer, it either expect it to be already installed or it installs at runtime using `brew install swiftlint`. The problem of this is that we can't predict which version of `swiftlint` is installed and therefore we don't know if it will work or not with the arguments and configuration we are running with. This situation leads to a potencial flaky behaviour increasing the maintenance burden.

What I want to explore in this PR is a different way to install `swiftlint` on the host computer and ensure that it will be always using a predicted version. Instead of relying on a global installed  `swiftlint` application I wanted to bundle the `swiftlint` application together with `danger-swiftlint` as a [gem extension](http://guides.rubygems.org/gems-with-extensions/). This way everytime the user install `danger-swiftlint` it will also be installing a particular version of `swiftlint` that comes with it.

As a consequence of pairing `danger-swiftlint` with `swiftlint` I think it makes sense for them to use the same version number. Meaning if you install `danger-swift@0.16.1` you will be using `swiftlint@0.16.1` everytime you run your dangerfile.

@ashfurrow I would be happy to hear your feedback if you think it makes sense.

Following is the list of tasks that I am expecting to tackle:
- [x] Sync danger-swiftlint and swiftlint versions
- [x] Update the gem to build swiftlint native extension
- [x] Update the plugin to use the extension interface to run `swiftlint`
- [x] Ensure tests are still passing

### CODE REVIEW

**Comment by thiagofelix** [comment]
With these changes when the user runs `gem install danger-swiftlint` during the build process gem will download the [portable_swiftlint.zip](https://github.com/realm/SwiftLint/releases/download/0.16.1/) file present on the release and store that close to the gem to be used at runtime.

Here is how the command output would look like:
```
gem install ./danger-swiftlint-0.16.1.gem
Building native extensions. 
Successfully installed danger-swiftlint-0.16.1
Parsing documentation for danger-swiftlint-0.16.1
Done installing documentation for danger-swiftlint after 0 seconds
1 gem installed
```

The line
**Building native extensions.  This could take a while...** is when it is downloading the portable zip file from the github releases page. After that danger-swiflint will have its own version of swiflint to use

File: lib/version.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 module DangerSwiftlint
-  VERSION = "0.3.0".freeze
+  VERSION = "0.16.1".freeze
```
Comment (ashfurrow) [inline lib/version.rb]: Is this meant to mirror the SwiftLint version?

Comment (thiagofelix) [inline lib/version.rb]: Yes it is, my goal to mirror both versions I believe is two fold 1) It is easy/quick for us to know which version of swiftlint will be used and what it is capable of and how much we need to "workaround" any limitation it has 2) From the user perspective they will be able to choose which version makes sense for them, perhaps they are not ready for a upgrade or they want to ensure always will use a given swiftlint version, etc, leading to a more predictable less flaky build.

Comment (thiagofelix) [inline lib/version.rb]: And of course, versions prior to `0.16.1` wouldn't benefit from this behavior and if you have `gem "0.3.0"` on your Gemfile that will still work as it is today and will download the latest `swiftlint` if it is not installed yet.

**Comment by thiagofelix** [comment]
Updated changelog @ashfurrow  =)

**Comment by thiagofelix** [comment]
Thanks for merging, looking forward to see how the new model performs into the wild. I'll extend the README to explain better the relationship of this library with swiftlint (versioning, isolated install, etc).

**Comment by ashfurrow** [comment]
Cool cool, thanks for contributing!

**Pull request merged**